File: backend/Services/Parsers/CostcodleParser.cs
using System.Text.RegularExpressions;

namespace backend.Services.Parsers
{
    public class CostcodleParser : BasicScoreResultParser
    {
        private ILogger<CostcodleParser> _logger;

        public CostcodleParser(ILogger<CostcodleParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override bool CountWinner => true;
        public override string GameName => "Costcodle";
        public override string? HelpText => null;
        protected override Regex Parser => new Regex($@"{GameName} #\d+ (?<{ScoreGroup}>[\d|X])/\d");
        protected override string? ExtraContent => Url;
        public override string Url => "https://costcodle.com";
    }
}

File: lambdas/src/parsers/costcodle-parser.ts
import { BasicScoreResultParser } from './base/basic-score-parser';

export class CostcodleParser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Costcodle';
  readonly helpText = null;
  readonly url = 'https://costcodle.com';
  protected readonly extraContent = this.url;
  
  // Matches "Costcodle #866 X/6"
  // Note: C# uses [\d|X] which incorrectly includes literal pipe - fixed here to [\dX]
  protected readonly parser = new RegExp(
    `${this.gameName} #\\d+ (?<${BasicScoreResultParser.SCORE_GROUP}>[\\dX])/\\d`
  );
  
  protected getCleanResult(result: string, match: RegExpMatchArray): string {
    return result
      .replace('https://costcodle.com/', '')
      .replace('https://costcodle.com', '')
      .trim();
  }
}
